### PR TITLE
Add planet drupal rss feed

### PR DESCRIPTION
**Pull request by richbrennan**
Have added ability to flag a blog post as being included in a new rss feed for inclusion on Planet Drupal

### CODE REVIEW

File: tbx/core/feeds.py [hunk 1]
```diff
@@ -29,3 +31,40 @@ def item_author_name(self, item):
 
     def item_pubdate(self, item):
         return datetime.combine(item.date, time())
+
+
+# Planet Drupal feed
+
+class PlanetDrupalFeed(Feed):
+    title = "The Torchbox Planet Drupal Feed"
+    link = "/planet_drupal_feed/"
```
Comment (zerolab) [inline tbx/core/feeds.py]: s/planet_drupal_feed/blog\/feed\/drupal/`

let's keep URLs consistent.

File: tbx/core/feeds.py [hunk 2]
```diff
@@ -29,3 +31,40 @@ def item_author_name(self, item):
...
+        return item.full_url
+
+    def item_author_name(self, item):
+        pass
+
+    def item_pubdate(self, item):
+        return datetime.combine(item.date, time())
+
+    def item_enclosure_url(self, item):
+        return 'http://127.0.0.1:8000' + item.feed_image.file.url
```
Comment (zerolab) [inline tbx/core/feeds.py]: shouldn't be localhost, no?

File: tbx/core/models.py [hunk 3]
```diff
@@ -553,6 +554,12 @@ class BlogPage(Page):
         related_name='+'
     )
     marketing_only = models.BooleanField(default=False, help_text='Display this blog post only on marketing landing page')
+    planet_drupal = models.BooleanField(
```
Comment (zerolab) [inline tbx/core/models.py]: We already use tags. the Drupal planet feed should really be blog posts filtered by the `drupal-planet` tag.

File: tbx/core/feeds.py [hunk 4]
```diff
@@ -29,3 +31,40 @@ def item_author_name(self, item):
...
+
+    def items(self):
+        return BlogPage.objects.live().filter(planet_drupal=True) \
+            .order_by('-date')
+
+    def item_title(self, item):
+        return item.title
+
+    def item_description(self, item):
+        return item.intro if item.intro else item.body
```
Comment (zerolab) [inline tbx/core/feeds.py]: This should include CTAGS, as the feed must validate to be included in the Drupal Planet.

See https://www.drupal.org/planet/guidelines (the `How to get your feed to the Planet` section)

**Comment by zerolab** [comment]
Cheers @richbrennan, starting to look good.

Added few comments and clarifications.

File: tbx/core/feeds.py (hunk 4 above)
Comment (richbrennan) [inline tbx/core/feeds.py]: I read the guidelines, no mention of CTAGS and the feed did validate so I'm not sure what you mean about CTAGS, could you elaborate?

Comment (zerolab) [inline tbx/core/feeds.py]: Argh, I meant [CDATA](https://developer.mozilla.org/en-US/docs/Web/RSS/Article/Why_RSS_Content_Module_is_Popular_-_Including_HTML_Contents) tags. From time to time one of the blog posts would throw the validator off.

Let's leave it out for now then.

**Comment by richbrennan** [comment]
@zerolab have pushed changes

**Comment by zerolab** [comment]
👍  from me

**Pull request merged**